lib/ansible/plugins/action/assert.py

```diff
@@ -38,12 +38,20 @@ def run(self, tmp=None, task_vars=None):
         if 'that' not in self._task.args:
             raise AnsibleError('conditional required in "that" string')
 
-        msg = None
+        fail_msg = None
         success_msg = None
-        if 'msg' in self._task.args and isinstance(self._task.args['msg'], string_types):
-            msg = self._task.args['msg']
-        if 'success_msg' in self._task.args and isinstance(self._task.args['success_msg'], string_types):
-            success_msg = self._task.args['success_msg']
+
+        fail_msg = self._task.args.get('fail_msg', self._task.args.get('msg'))
+        if fail_msg is None:
+            fail_msg = 'Assertion failed'
+        if not isinstance(fail_msg, string_types):
```

Comment:
not a blocker, but a small optimization, `elif` will avoid the check if fail_msg was `None`, since you are hardcoding to string you should not need to do this